src/MessageHistoryTable/MessageHistoryRow.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import styled from 'styled-components'
+import { bool, string } from 'prop-types'
+import { MESSAGE_PROPS, ADDRESS_PROPTYPE } from '../customPropTypes'
+import Box from '../Box'
+import { Menu, MenuItem } from '../Menu'
+import { Text } from '../Typography'
+import { IconSend, IconReceive } from '../Icons'
+
+const MessageHistoryRowContainer = styled(Box)`
+  &:nth-child(even) {
+    transform: translateY(-1px);
+  }
+`
+
+const AddressText = ({ sentMsg, to, from }) => {
+  if (sentMsg) {
+    return (
+      <>
+        <Text my={0}>To</Text>
+        <Text mt={2} mb={0}>
+          {to}
+        </Text>
+      </>
+    )
+  }
+
+  return (
+    <>
+      <Text my={0}>From</Text>
+      <Text mt={2} mb={0}>
+        {from}
+      </Text>
+    </>
+  )
+}
+
+AddressText.propTypes = {
+  sentMsg: bool.isRequired,
+  to: ADDRESS_PROPTYPE,
+  from: ADDRESS_PROPTYPE
+}
+
+const ActionText = ({ status, sentMsg }) => {
+  if (status === 'confirmed' && sentMsg) return <Text>Sent</Text>
+  else if (status === 'confirmed') return <Text>Received</Text>
+  else if (status === 'pending' && sentMsg) return <Text>Sending</Text>
+  // an unconfirmed received  sg
+  else if (status === 'pending') return <Text>Confirming</Text>
```

Comment:
Couldn't we just use a single `Text` element and wrap the conditional statement inside of it, so it only renders the text rather than the entire element?